src/time.cpp

```diff
@@ -137,27 +186,38 @@ void fields::output_times(const char *fname) {
   FILE *tf = master_fopen(fname, "w");
   if (!tf) meep::abort("Unable to create file %s!\n", fname);
 
-  int n = count_processors();
-  double *alltimes_tmp = new double[n * (Other + 1)];
-  double *alltimes = new double[n * (Other + 1)];
-  for (int i = 0; i <= Other; ++i) {
-    for (int j = 0; j < n; ++j)
-      alltimes_tmp[i * n + j] = j == my_rank() ? times_spent[i] : 0;
-  }
-  sum_to_master(alltimes_tmp, alltimes, n * (Other + 1));
-  delete[] alltimes_tmp;
+  std::vector<double> alltimes = timing_data_vector_from_all(times_spent);
 
-  for (int i = 0; i <= Other-1; ++i)
-    master_fprintf(tf, "%s, ", ts2n((time_sink)i));
-  master_fprintf(tf, "%s\n", ts2n(Other));
+  const char *sep = "";
+  for (const auto &desc_ts : kDescriptionByTimeSink) {
+    master_fprintf(tf, "%s%s", sep, desc_ts.second);
+    sep = ", ";
+  }
+  master_fprintf(tf, "\n");
 
+  const int n = count_processors();
   for (int j = 0; j < n; ++j) {
-    for (int i = 0; i <= Other-1; ++i)
-      master_fprintf(tf, "%g, ", alltimes[i * n + j]);
-    master_fprintf(tf, "%g\n", alltimes[Other * n + j]);
+    const char *sep = "";
+    for (size_t i = 0; i < kDescriptionByTimeSink.size(); ++i) {
+      master_fprintf(tf, "%s%g", sep, alltimes[i * n + j]);
+      sep = ", ";
+    }
+    master_fprintf(tf, "\n");
   }
   master_fclose(tf);
-  delete[] alltimes;
+}
+
+std::unordered_map<time_sink, std::vector<double> > fields::get_timing_data() const {
```

Comment:
see above comment about enums as `unordered_map` keys

Reply:
Not applicable.

Reply:
Apple clang is giving me an error:
```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/unordered_map:427:35: error: no member named 'value' in 'std::__1::is_empty<std::__1::hash<meep::time_sink>>'
          bool = is_empty<_Hash>::value && !__libcpp_is_final<_Hash>::value>
                 ~~~~~~~~~~~~~~~~~^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/unordered_map:862:13: note: in instantiation of default argument for '__unordered_map_hasher<meep::time_sink, std::__1::__hash_value_type<meep::time_sink, std::__1::vector<double>>, std::__1::hash<meep::time_sink>>' required here
    typedef __unordered_map_hasher<key_type, __value_type, hasher>   __hasher;
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
time.cpp:210:61: note: in instantiation of template class 'std::__1::unordered_map<meep::time_sink, std::__1::vector<double>>' requested here
std::unordered_map<time_sink, std::vector<double> > fields::get_timing_data() const {
                                                            ^
In file included from time.cpp:18:
In file included from ./meep.hpp:23:
```

Reply:
Fixed.